src/create_oauth2_client.ts

```diff
@@ -1,360 +1,16 @@
-  additionalOAuth2ClientConfig:
-    & Partial<OAuth2ClientConfig>
-    & WithRedirectUri
-    & WithScope,
-): OAuth2Client {
-  return new OAuth2Client({
-    clientId: Deno.env.get("PATREON_CLIENT_ID")!,
-    clientSecret: Deno.env.get("PATREON_CLIENT_SECRET")!,
-    authorizationEndpointUri: "https://www.patreon.com/oauth2/authorize",
-    tokenUri: "https://www.patreon.com/api/oauth2/token",
-    ...additionalOAuth2ClientConfig,
-  });
-}
-
-/**
- * Creates an OAuth 2.0 client with Slack as the provider.
- *
- * Requires `--allow-env[=SLACK_CLIENT_ID,SLACK_CLIENT_SECRET]` permissions and environment variables:
- * 1. `SLACK_CLIENT_ID`
- * 2. `SLACK_CLIENT_SECRET`
- *
- * @param additionalOAuth2ClientConfig Requires `defaults.scope` property.
- *
- * @example
- * ```ts
- * import { createSlackOAuth2Client } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
- *
- * const oauth2Client = createSlackOAuth2Client({
- *  redirectUri: "http://localhost:8000/callback",
- *  defaults: {
- *    scope: "users.profile:read"
- *  }
- * });
- * ```
- *
- * @see {@link https://api.slack.com/authentication/oauth-v2}
- */
-export function createSlackOAuth2Client(
-  additionalOAuth2ClientConfig: Partial<OAuth2ClientConfig> & WithScope,
-): OAuth2Client {
-  return new OAuth2Client({
-    clientId: Deno.env.get("SLACK_CLIENT_ID")!,
-    clientSecret: Deno.env.get("SLACK_CLIENT_SECRET")!,
-    authorizationEndpointUri: "https://slack.com/oauth/v2/authorize",
-    tokenUri: "https://slack.com/api/oauth.v2.access",
-    ...additionalOAuth2ClientConfig,
-  });
-}
-
-/**
- * Creates an OAuth 2.0 client with Twitter as the provider.
- *
- * Requires `--allow-env[=TWITTER_CLIENT_ID,TWITTER_CLIENT_SECRET]` permissions and environment variables:
- * 1. `TWITTER_CLIENT_ID`
- * 2. `TWITTER_CLIENT_SECRET`
- *
- * @param additionalOAuth2ClientConfig Requires `redirectUri` and `defaults.scope` properties.
- *
- * @example
- * ```ts
- * import { createTwitterOAuth2Client } from "https://deno.land/x/deno_kv_oauth@$VERSION/mod.ts";
- *
- * const oauth2Client = createTwitterOAuth2Client({
- *  redirectUri: "http://localhost:8000/callback",
- *  defaults: {
- *    scope: "users.read"
- *  }
- * });
- * ```
- *
- * @see {@link https://github.com/twitterdev/twitter-api-typescript-sdk}
- */
-export function createTwitterOAuth2Client(
-  additionalOAuth2ClientConfig:
-    & Partial<OAuth2ClientConfig>
-    & WithRedirectUri
-    & WithScope,
-): OAuth2Client {
-  return new OAuth2Client({
-    clientId: Deno.env.get("TWITTER_CLIENT_ID")!,
-    clientSecret: Deno.env.get("TWITTER_CLIENT_SECRET")!,
-    authorizationEndpointUri: "https://twitter.com/i/oauth2/authorize",
-    tokenUri: "https://api.twitter.com/2/oauth2/token",
-    ...additionalOAuth2ClientConfig,
-  });
-}
+export type WithScope = { defaults: { scope: string | string[] } };
+export type WithRedirectUri = { redirectUri: string };
+
+export { createDiscordOAuth2Client } from "./providers/discord.ts";
```

Comment:
```suggestion
export * from "./providers/discord.ts";
```
I think it's fine to do this for all of these.